katcp/client.py

```diff
@@ -852,32 +860,81 @@ def join(self, timeout=None):
 
         Notes
         -----
-        Does nothing if the ioloop is not managed.
+        Does nothing if the ioloop is not managed.  Use until_stopped() instead.
 
         """
         self._ioloop_manager.join(timeout)
 
     def stop(self, timeout=None):
-        """Stop a running client (from another thread).
+        """Stop a running client.
+
+        If using a managed ioloop, this must be called from a different
+        thread to the ioloop's.  This method only returns once the client's
+        main coroutine, `_install()`, has completed.
+
+        If using an unmanaged ioloop, this can be called from the
+        same thread as the ioloop.  The `until_stopped()` method can be
+        used to wait on completion of the main coroutine, `_install()`.
 
         Parameters
         ----------
         timeout : float in seconds
-           Seconds to wait for client thread to have *started*.
+           Seconds to wait for both client thread to have *started*, and
+           for stopping.
```

Comment:
What happens if the client thread doesn't stop in time? Is an exception raised?

Reply:
Doesn't look like an exception is raised, so caller won't know.  Guess code might get stuck when waiting on `join` or `until_stopped`.  This is how it was working before, so I don't plan on fixing it.